### PR TITLE
Adds dedicated turn lane going South.

### LINKED ISSUE #27: Map request: Dedicated turn lane going south
### Map request

image

### Implementation suggestion

 it's okay to include a median between the two South lanes
 
 
image

### PR DESCRIPTION
**Pull request by agalbachicar**
# 🎉 New feature

Closes #27

## Summary
Implements 

image

## Test it

To try it, run:

```sh
WIDTH=3.3 STOPLINE=True RADIUS=6.0 empy3 src/maliput_xodr/templates/dedicated_turn_lane_going_south.xml.em
```

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by liangfok** [comment]
Thanks for creating this! I'll test it ASAP.

**Comment by liangfok** [comment]
@agalbachicar I am running into the following error when loading this map in `maliput_malidrive`:

```
terminate called after throwing an instance of 'maliput::common::assertion_error'
  what():  db_manager.cc:VerifyJunctions:604: When verifying JunctionId: 10 , the junction of the connecting road 9 is: -1 which does not match the expected junction id: 10
```
Here's the line of code that is throwing the exception:
https://github.com/maliput/maliput_malidrive/blob/6bdc024e0dfa8547107f902b9a6825f7217c7326/src/maliput_malidrive/xodr/db_manager.cc#L599-L605

**Comment by francocipollone** [comment]
CC: @agalbachicar 

using `xodr_validate` is useful to find issues in the xodr file. 
`xodr_validate --xodr_file=<path> --log_level=trace`

I've used it a lot for debugging, let me take a look asap

**Comment by agalbachicar** [comment]
Thanks both for pointing it out. I'll get a fixed version ASAP.

**Comment by agalbachicar** [comment]
Should be fixed now @liangfok , PTAL.

**Comment by liangfok** [comment]
I confirm the new map is loadable in our downstream project. We're currently testing it. Thank you!

**Comment by liangfok** [comment]
Initial testing revealed two issues: (1) missing stop sign, and (2) intersection should be T-shaped rather than 4-way.

(1) is strange since I see stop lines in the .geojson file. @vladimir-wp, is anything else needed for stop signs to appear in our downstream map format?

(2) was due to changes in the specification over time. The original version was 4-way, but a subsequent iteration changed it to be T-shaped. Below is the version we'd like to model. Note that it is T-shaped and there is only one stop sign.

T-shaped-Intersection

**Comment by liangfok** [comment]
I coordinated with @vladimir-wp and determined that the GeoJSON needs to have another new boolean key/value pair that specifies whether a stop line has an associated stop sign. Instead of:

```
      "properties": {
        ...
        "Type": "Stopline",
        ...
      },
```

it should have:

```
      "properties": {
        ...
        "Type": "Stopline",
        "HasStopSign": "True",
        ...
      },
```

This will enable our downstream tooling to model maps containing a mixture of stop lines, some with and some without stop signs.

The default value of "HasStopSign" will be "True" so for the map in this PR, the new "HasStopSign" key can be omitted.

@agalbachicar can you update the .xodr & GeoJSon to be T-shaped and while @vladimir-wp works on updating our downstream tooling to support the new "HasStopSign" key?

**Comment by francocipollone** [comment]
> I coordinated with @vladimir-wp and determined that the GeoJSON needs to have another new boolean key/value pair that specifies whether a stop line has an associated stop sign. Instead of:
> 
> ```
>       "properties": {
>         ...
>         "Type": "Stopline",
>         ...
>       },
> ```
> 
> it should have:
> 
> ```
>       "properties": {
>         ...
>         "Type": "Stopline",
>         "HasStopSign": "True",
>         ...
>       },
> ```
> 
> This will enable our downstream tooling to model maps containing a mixture of stop lines, some with and some without stop signs.
> 
> The default value of "HasStopSign" will be "True" so for the map in this PR, the new "HasStopSign" key can be omitted.
> 
> @agalbachicar can you update the .xodr & GeoJSon to be T-shaped and while @vladimir-wp works on updating our downstream tooling to support the new "HasStopSign" key?

Noted! I will add the fixes

**Comment by liangfok** [comment]
Many thanks!!

**Comment by liangfok** [comment]
Would it be possible to get an updated version of this map early next week? Our downstream users wish to have it integrated into their test suite by end of next week (Sept. 30).

**Comment by francocipollone** [comment]
> Would it be possible to get an updated version of this map early next week? Our downstream users wish to have it integrated into their test suite by end of next week (Sept. 30).

Definitely!

**Comment by liangfok** [comment]
Thank you!!

**Comment by francocipollone** [comment]
Updated intersection according to requirements:

image

File: resources/dedicated_turn_lane_going_south/dedicated_turn_lane_going_south.xodr.geojson [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            37.417856413454345,
+            0.0
+          ]
+          ],
+        "type": "LineString"
+      },
+      "properties": {
+        "Id": "{ba11d4ae-3784-11ee-be56-0242ac120002}",
+        "Type": "Stopline",
+        "HasStopSign": "True"
```
Comment (francocipollone) [inline resources/dedicated_turn_lane_going_south/dedicated_turn_lane_going_south.xodr.geojson:23]: Also added the new `HasStopSign` label. CC: @liangfok

**Comment by francocipollone** [comment]
@liangfok please take a look

**Comment by francocipollone** [comment]
CC: @agalbachicar

**Comment by liangfok** [comment]
LGTM. I confirmed this map is sufficient for our downstream users. Eventually, we may want to implement the transitions that go from the stop sign, through the intersection, to the southbound lane, but that is not needed at this time.

**Review by liangfok — APPROVED** [review]
LGTM

**Pull request merged**